sql-plugin/src/main/scala/org/apache/spark/sql/rapids/complexTypeCreator.scala

```diff
@@ -106,7 +106,23 @@ case class GpuCreateMap(children: Seq[Expression], useStringTypeWhenEmpty: Boole
       val structs = Range(0, columns.length, 2)
         .safeMap(i => ColumnVector.makeStruct(columns(i), columns(i + 1)))
       withResource(structs) { _ =>
-        GpuColumnVector.from(ColumnVector.makeList(numRows, DType.STRUCT, structs: _*), dataType)
+        withResource(ColumnVector.makeList(numRows, DType.STRUCT, structs: _*)) { listOfStruct =>
+          withResource(listOfStruct.dropListDuplicatesWithKeysValues()) { deduped =>
+            if (SQLConf.get.getConf(SQLConf.MAP_KEY_DEDUP_POLICY) ==
+                SQLConf.MapKeyDedupPolicy.EXCEPTION.toString) {
```

Comment:
Nit: Rather than perform a conf lookup for each columnar batch, this should either be cached as a lazy val of this class or looked up once on the driver and passed in as a class parameter.